src/Engine/ProtoCore/Namespace/ElementRewriter.cs

```diff
@@ -1,26 +1,24 @@
-﻿using System.Diagnostics;
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
 using ProtoCore.AST;
 using ProtoCore.AST.AssociativeAST;
 using ProtoCore.DSASM;
 using ProtoCore.SyntaxAnalysis;
 using ProtoCore.Utils;
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
 
 namespace ProtoCore.Namespace
 {
 
     public class ElementRewriter : AstReplacer
     {
-        private readonly ClassTable classTable;
+        private readonly Core core;
         private readonly ElementResolver elementResolver;
 
-        internal ElementRewriter(ClassTable classTable, ElementResolver resolver)
+        public ElementRewriter(Core core, ElementResolver resolver = null)
```

Comment:
It would be good if you could avoid the dependency on Core. You can use event handlers to register for Build status logging.